test.js

```diff
@@ -185,6 +198,126 @@ describe('cli', function () {
     })
   })
 
+  describe('pre-release', function () {
+    it('works fine without specifying a tag id when prereleasing', function () {
+      writePackageJson('1.0.0')
+      fs.writeFileSync('CHANGELOG.md', 'legacy header format<a name="1.0.0">\n', 'utf-8')
+
+      commit('feat: first commit')
+      return execCliAsync('--prerelease')
+        .then(function () {
+          // it's a feature commit, so it's minor type
+          getPackageVersion().should.equal('1.1.0-0')
+        })
+    })
+  })
+
+  describe('manual-release', function () {
+    it('throws error when not specifying a release type', function () {
+      writePackageJson('1.0.0')
+      fs.writeFileSync('CHANGELOG.md', 'legacy header format<a name="1.0.0">\n', 'utf-8')
+
+      commit('fix: first commit')
+      execCli('--release-as').code.should.above(0)
+    })
+
+    describe('release-types', function () {
+      var regularTypes = ['major', 'minor', 'patch']
+
+      regularTypes.forEach(function (type) {
```

Comment:
But in this case we dont add a bunch of fixture to avoid duplication. I think its readability is acceptable, just opinion. It you insist, i can expand it. I would be much longer.